### DISCUSSION TITLE
Emoji flags handling seems a bit off

### DISCUSSION THREAD

**Discussion by unorsk** [ghostty-org/ghostty#3101] [Q&A] [answered]
I am not sure if this is a problem with ghostty, but I experienced this using zsh, bash and vim.

Steps to reproduce:
* Just input echo "🇨🇭🇵🇱🇩🇰" in the terminal :)

And then what I am getting is somehow unpredictable behaviour – some of the symbols disappear, until I hit "return" and it prints everything correctly.
But then when I use "up" command to get to my previous command and remove a random flag, it gets weird again – once I remove a flag in the middle, the right one renders incorrectly.

https://github.com/user-attachments/assets/aaf5b5d8-2537-4407-ba06-aca7b4136030

https://github.com/user-attachments/assets/8585ae88-805f-4821-ad74-512cf81ed56b

I'm running MacOS 15.1.1 (24B91)

<img width="412" alt="Image" src="https://github.com/user-attachments/assets/377408ff-331c-48a8-a5f6-7915c1130692" />

**Comment by mitchellh** [comment, marked as answer]
Zsh bash and vim all do not handle grapheme clusters correctly. This is a bug with them. For more details see my blog: https://mitchellh.com/writing/grapheme-clusters-in-terminals

**Reply by peterjc** [reply]
Ah - I was hoping this was a terminal bug, e.g. WezTerm issue https://github.com/wez/wezterm/issues/6364 which happens in Ghostty too. It makes sense that this is a problem in zsh and bash etc 😞
